Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -493,12 +493,18 @@ void SetupIDs()
                     // force 0 for prefabs
                     sceneId = 0;
                     //Debug.Log(name + " @ scene: " + gameObject.scene.name + " sceneid reset to 0 because CurrentPrefabStage=" + PrefabStageUtility.GetCurrentPrefabStage() + " PrefabStage=" + PrefabStageUtility.GetPrefabStage(gameObject));
-                    // NOTE: might make sense to use GetPrefabStage for asset
-                    //       path, but let's not touch it while it works.
 #if UNITY_2020_1_OR_NEWER
                     string path = PrefabStageUtility.GetCurrentPrefabStage().assetPath;
+
+                    // Try GetPrefabStage if empty path, when GetCurrentPrefabStage doesn't work (Unity wierdness?)
```

Comment:
I still don't understand how you got to this change.
did you just try the other function and it magically worked?
or was there a fundamental reason why we should use GetPrefabStage over GetCurrentPrefabStage?
in fact, should we not always use GetPrefabStage over GetCurrentPrefabStage?
and is there a way to reproduce the issue?

Reply:
I put this in based on your comment, as a fallback if GetCurrentPrefabStage doesn't.  Both of these are in Unity's experimental namespace, and largely undocumented as to the difference between them.  I'm just trying to make this robust.

Reply:
could you open an issue with a way to reproduce the failure first please?
we should be careful with adding magic that we don't understand.

if there's an issue we can reproduce, then we can work from there

Reply:
I have no idea under what circumstance one may work and not the other.  Neither are documented.  I put this in there based on your original comment note.